src/test/descriptor_tests.cpp

```diff
@@ -40,32 +43,46 @@ bool EqualDescriptor(std::string a, std::string b)
     return a == b;
 }
 
-std::string MaybeUseHInsteadOfApostrophy(std::string ret)
+std::string UseHInsteadOfApostrophe(std::string ret)
 {
-    if (InsecureRandBool()) {
-        while (true) {
-            auto it = ret.find("'");
-            if (it != std::string::npos) {
-                ret[it] = 'h';
-                if (ret.size() > 9 && ret[ret.size() - 9] == '#') ret = ret.substr(0, ret.size() - 9); // Changing apostrophe to h breaks the checksum
-            } else {
-                break;
+    while (true) {
+        auto it = ret.find("'");
+        if (it != std::string::npos) {
+            ret[it] = 'h';
+            if (ret.size() > 9 && ret[ret.size() - 9] == '#') {
+                Span<const char> data = Span<const char>(ret.data(), ret.size() - 9);
+                std::string new_checksum = DescriptorChecksum(data);
+                ret = ret.substr(0, ret.size() - 8) + new_checksum; // Changing apostrophe to h breaks the checksum - recalculate it and replace the broken one
             }
+        } else {
+            break;
         }
     }
     return ret;
 }
 
 const std::set<std::vector<uint32_t>> ONLY_EMPTY{{}};
 
-void Check(const std::string& prv, const std::string& pub, int flags, const std::vector<std::vector<std::string>>& scripts, const std::set<std::vector<uint32_t>>& paths = ONLY_EMPTY)
+void DoCheck(const std::string& prv, const std::string& pub, int flags, const std::vector<std::vector<std::string>>& scripts, const std::set<std::vector<uint32_t>>& paths = ONLY_EMPTY,
+    bool replace_apostrophe_with_h=false)
 {
     FlatSigningProvider keys_priv, keys_pub;
     std::set<std::vector<uint32_t>> left_paths = paths;
 
+    std::unique_ptr<Descriptor> parse_priv;
+    std::unique_ptr<Descriptor> parse_pub;
     // Check that parsing succeeds.
-    auto parse_priv = Parse(MaybeUseHInsteadOfApostrophy(prv), keys_priv);
-    auto parse_pub = Parse(MaybeUseHInsteadOfApostrophy(pub), keys_pub);
+    if (replace_apostrophe_with_h)
+    {
+        parse_priv = Parse(prv, keys_priv);
+        parse_pub = Parse(pub, keys_pub);
+    }
+    else
+    {
+        parse_priv = Parse(UseHInsteadOfApostrophe(prv), keys_priv);
+        parse_pub = Parse(UseHInsteadOfApostrophe(pub), keys_pub);
```

Comment:
This is actually backwards. When `replace_apostrophe_with_h` is false, we're swapping. I think it's supposed to be the other way around.

Reply:
Thanks, you're absolutely right. Fixed in a8f49b7d53d33aeba8bd0965c21878795b36f99b.